zrml/prediction-markets/src/migrations.rs

```diff
@@ -1 +1,216 @@
+use crate::{
+    Config, Disputes, LastTimeFrame, MarketIdsPerCloseBlock, MarketIdsPerCloseTimeFrame, MomentOf,
+    Pallet,
+};
+use alloc::vec;
+use frame_support::{
+    dispatch::Weight,
+    log,
+    pallet_prelude::PhantomData,
+    traits::{Get, OnRuntimeUpgrade, StorageVersion},
+};
+use zeitgeist_primitives::{
+    traits::Swaps as SwapsPalletApi,
+    types::{Market, MarketPeriod, MarketStatus, PoolStatus},
+};
+use zrml_market_commons::MarketCommonsPalletApi;
 
+const PREDICTION_MARKETS_REQUIRED_STORAGE_VERSION: u16 = 1;
+const PREDICTION_MARKETS_NEXT_STORAGE_VERSION: u16 = 2;
+
+pub struct RemoveDisputesOfResolvedMarkets<T>(PhantomData<T>);
+
+impl<T: Config> OnRuntimeUpgrade for RemoveDisputesOfResolvedMarkets<T> {
+    fn on_runtime_upgrade() -> Weight
+    where
+        T: Config,
+    {
+        let mut total_weight = T::DbWeight::get().reads(1);
+
+        if StorageVersion::get::<Pallet<T>>() != PREDICTION_MARKETS_REQUIRED_STORAGE_VERSION {
+            log::info!(
+                "Skipping storage removal of disputes of resolved markets; prediction-markets \
+                 already up to date"
+            );
+            return total_weight;
+        }
+        total_weight = total_weight.saturating_add(T::DbWeight::get().reads(1));
+        log::info!("Starting storage migration of RemoveDisputesOfResolvedMarkets");
+
+        for (market_id, market) in T::MarketCommons::market_iter() {
+            total_weight = total_weight.saturating_add(T::DbWeight::get().reads(1));
+
+            match market.status {
+                MarketStatus::Resolved => {
+                    Disputes::<T>::remove(market_id);
```

Comment:
```suggestion
                    Disputes::<T>::remove(market_id);
                    total_weight = total_weight.saturating_add(T::DbWeight::get().writes(1));
```

Isn't this a storage write?